pysindy/optimizers/constrained_sr3.py

```diff
@@ -205,6 +233,41 @@ def _update_full_coef_constraints(self, H, x_transpose_y, coef_sparse):
         rhs = rhs.reshape(g.shape)
         return inv1.dot(rhs)
 
+    def _update_coef_cvxpy(self, x, y, coef_sparse):
+        xi = cp.Variable(coef_sparse.shape[0] * coef_sparse.shape[1])
+        cost = cp.sum_squares(x @ xi - y.flatten())
+        if self.thresholder.lower() == "l1":
+            cost = cost + self.threshold * cp.norm1(xi)
+        elif self.thresholder.lower() == "weighted_l1":
+            cost = cost + cp.norm1(self.threshold * xi)
+        elif self.thresholder.lower() == "l2":
+            cost = cost + self.threshold * cp.norm2(xi)
+        elif self.thresholder.lower() == "weighted_l2":
+            cost = cost + cp.norm2(self.thresholds * xi)
```

Comment:
Same comment as above - I think l2 is not implemented for SR3?

Reply:
See above. I put in the prox and regularization in /utils/base.py for L2, so it is compatible.